sensor_module/sensor_module/src/src/intercomm.c

```diff
@@ -9,111 +9,99 @@
 
 #include "intercomm.h"
 #include "spi.h"
+#include "sensor_data.h"
+#include "utilities.h"
 
-#define MAX_ALIGN_ATTEMPTS 5
-#define MAX_COMM_ATTEMPTS 5
-
+// Forward declares
 void send_status_and_data_packet(void);
+void send_sensor_data_and_update_check_byte(uint8_t sensor_data_id, uint8_t* check_byte);
+void send_and_update_check_byte(uint8_t data, uint8_t* check_byte);
+
 
 static uint8_t spi_aligned = 0;
 static uint8_t spi_read = 0;
-static uint8_t spi_finished = 0;
+static uint8_t spi_confirmed = 0;
 
 
-/* Temporary structure of data to send, will be updated. */
-static uint8_t check_byte = 0;
-
+// Temporary static variables
 static uint8_t status_1 = 0x01;
 static uint8_t status_2 = 0x02;
 
-static uint8_t acc_x_1 = 0x03;
-static uint8_t acc_x_2 = 0x04;
-static uint8_t acc_y_1 = 0x05;
-static uint8_t acc_y_2 = 0x06;
-static uint8_t acc_z_1 = 0x07;
-static uint8_t acc_z_2 = 0x08;
-
-static uint8_t dist = 0x09;
-
-static uint8_t speed = 0x0A;
-
 // SPI Transmission/reception completed interrupt service routine
 ISR(SPI_STC_vect)
 {	
-	uint8_t comm_attempts = 0;
-	// Attempt a full communication until finished or it has been attempted over max amount
-	while (spi_finished == 0 && comm_attempts < MAX_COMM_ATTEMPTS) {
+	spi_read = spi_get_data_register_value();
+
+	spi_aligned = (spi_read == SPI_START); // SPI is aligned if START was read
+	if (spi_aligned) {
+		// Send SPI acknowledge byte
+		spi_transcieve(SPI_ACK);
 		
-		uint8_t align_attempts = 0;
-		// Attempt to align SPI until aligned or until it has been attempted over max amount
-		while (spi_aligned == 0 &&  align_attempts < MAX_ALIGN_ATTEMPTS) {
-			spi_read = SPDR;
-			PORTA = spi_read;
-			spi_aligned = spi_read == SPI_START; // SPI is aligned if START was read
-			if (spi_aligned) {
-				spi_transcieve(SPI_ACK);
-			
-			} else {
-				spi_transcieve(SPI_RESTART);
-			}
-			
-			align_attempts++;
+		if (SPSR & (1 << WCOL))
+		{
+			utilities_error(0x12);
```

Comment:
Vi borde ta bort error här och istället avsluta överföringen så att den får starta om. Sannolikheten att vi går in här är väldigt låg, men gör vi det slutar sensormodulen att svara helt eftersom error har en oändlig loop.

Reply:
Jo sannt, vi gör så